### PR TITLE
eventlog: Add handling of sha1 log format (Fixes #2179).

### LINKED ISSUE #2179: eventlog parse failure on HP firmware
Any idea what is going on here? 

The binary_bios_measurements_parser.py script can parse this but tpm2_eventlog cannot. 

[root@localhost ~]# cat /sys/kernel/security/tpm0/binary_bios_measurements > log.bin
[root@localhost ~]# tpm2_eventlog log.bin
---
ERROR: SpecID eventType must be EV_NO_ACTION
ERROR: failed to parse tpm2 eventlog
ERROR: Unable to run tpm2_eventlog

[root@localhost ~]# fwupdmgr get-devices
HP ProBook 650 G5 System Firmware
  DeviceId:             82c0b707f7573e30fa548ff4ea37b53f4c9b5bf6
  Guid:                 dd29e2bb-b1f4-4c14-9f4d-9848d7fd5e09
  Guid:                 68264ab0-73cb-5572-a217-6ba4482083d2
  Guid:                 230c8b18-8d9b-53ec-838b-6cfc0383493a
  Plugin:               uefi
  Flags:                internal|updatable|require-ac|registered|needs-reboot
  Version:              0.1.769
  VersionLowest:        0.0.1
  Icon:                 computer
  Created:              2020-08-22

### PR DESCRIPTION
**Pull request by JuergenReppSIT**
If no specid event is part of the event list no description of the crytpo agile event
format is available. In this case it will be assumed that the sha1 log format is
used.
A function for parsing the sha1 log event header and yaml callbacks for this format
were added.

Signed-off-by: Juergen Repp <juergen.repp@sit.fraunhofer.de>

### CODE REVIEW

**Comment by idesai** [comment]
@JuergenReppSIT can you add a test binary to the fixtures and update the test?

File: lib/tpm2_eventlog.c [hunk 1]
```diff
@@ -196,6 +196,85 @@ bool parse_event2(TCG_EVENT_HEADER2 const *eventhdr, size_t buf_size,
...
+    return true;
+}
+
+bool foreach_sha1_log_event(tpm2_eventlog_context *ctx, TCG_EVENT const *eventhdr_start, size_t size) {
+
+    if (eventhdr_start == NULL) {
+        LOG_ERR("invalid parameter");
+        return false;
+    }
+    if (size == 0) {
```
Comment (idesai) [inline lib/tpm2_eventlog.c:233]: new line here

File: lib/tpm2_eventlog.c [hunk 2]
```diff
@@ -340,19 +419,26 @@ bool parse_eventlog(tpm2_eventlog_context *ctx, BYTE const *eventlog, size_t siz
     TCG_EVENT *event = (TCG_EVENT*)eventlog;
     bool ret;
 
-    ret = specid_event(event, size, &next);
-    if (!ret) {
+    if (!event)
```
Comment (idesai) [inline lib/tpm2_eventlog.c:422]: Add the braces for the dependent block

File: lib/tpm2_eventlog.h [hunk 3]
```diff
@@ -16,10 +16,14 @@ typedef bool (*EVENT2_CALLBACK)(TCG_EVENT_HEADER2 const *event_hdr, size_t size,
...
                                     void *data);
 typedef bool (*SPECID_CALLBACK)(TCG_EVENT const *event, void *data);
+typedef bool (*LOG_EVENT_CALLBACK)(TCG_EVENT const *event_hdr, size_t size,
+                                   void *data);
+
 
 typedef struct {
     void *data;
     SPECID_CALLBACK specid_cb;
+     LOG_EVENT_CALLBACK log_eventhdr_cb;
```
Comment (idesai) [inline lib/tpm2_eventlog.h:26]: Remove leading whitespace

File: lib/tpm2_eventlog.c [hunk 4]
```diff
@@ -340,19 +419,26 @@ bool parse_eventlog(tpm2_eventlog_context *ctx, BYTE const *eventlog, size_t siz
...
+            ret = ctx->specid_cb(event, ctx->data);
+            if (!ret) {
+                return false;
+            }
+        }
+        return foreach_event2(ctx, next, size);
+      } else {
+        /* No specid event found. sha1 log format will be parsed. */
+        return foreach_sha1_log_event(ctx, event, size);
+    }
```
Comment (idesai) [inline lib/tpm2_eventlog.c:443]: Closing brace of the first and second dependent block looks misaligned.

**Review by idesai — CHANGES_REQUESTED** [review]

**Comment by JuergenReppSIT** [comment]
I have added the test binary from the issue to the fixtures.

**Pull request merged**